python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -0,0 +1,309 @@
+import heapq
+import random
+from typing import Any, Callable, Iterator, List, Optional, Tuple
+
+from ray.types import ObjectRef
+from ray.data.block import Block, BlockMetadata, BlockAccessor, DataBatch
+from ray.data._internal.block_batching.interfaces import (
+    Batch,
+    LogicalBatch,
+    BlockPrefetcher,
+)
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.memory_tracing import trace_deallocation
+
+
+def _bundle_block_refs_to_logical_batches(
```

Comment:
Note, don't need to use underscore methods as this entire module is _internal.

Reply:
sure, updated